harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -184,6 +195,33 @@ def download(self, path: Optional[str] = None) -> str:
 
         return str(local_ckpt_dir)
 
+    def _download_via_master(self, local_ckpt_dir: pathlib.Path) -> None:
+        local_ckpt_dir.mkdir(parents=True, exist_ok=True)
+        if os.name == "nt":
+            fext = "zip"
+            mime_type = "application/zip"
+        else:
+            fext = "tgz"
+            mime_type = "application/gzip"
+        resp = self._session.get(f"/checkpoints/{self.uuid}", headers={"Accept": mime_type})
+        if not resp.ok:
+            raise RuntimeError(
+                "unable to download checkpoint from master:", resp.status_code, resp.reason
+            )
+
+        tmpfile = pathlib.Path(f"{local_ckpt_dir}/{self.uuid}.{fext}")
```

Comment:
I like using a file for temporary storage, but can we use `tempfile.mkstemp()` instead?  (see notes [here](https://github.com/determined-ai/determined/blob/master/harness/determined/cli/shell.py#L106) as to why it's valuable)

Reply:
Actually, hm... maybe if you set `stream=True` on the `session.get()` you might be able to stream straight from the network into the tar extraction.  Let me poke around a bit.

Reply:
Looks like you have to [make your own file-like object](https://github.com/psf/requests/issues/465#issuecomment-7276058) to do it, but you would really decrease disk io costs.  That's worth it I think, while we are building something to download multiple GB, especially when our server doesn't support resuming downloads on failures anyway.

Reply:
Yeah, I thought about that but didn't dig enough. Let me take a look.